content/filterx/_index.md

```diff
@@ -0,0 +1,374 @@
+
+Like simple filters, filterx blocks consist of a list of filter statements. If a message matches all filter statements, it passes the filter.
+
+The result of filterx statements is always a boolean value, either *true* or *false*:
+
+- True values are: Complex values, non-empty strings, and non-zero numbers.
+- False values are:
+    - empty strings,
+    - the `false` value,
+    - the `0` value,
+    - statements that result in an error (for example, if a comparison cannot be evaluated because of type error, or a field of a variable referenced in the statement is doesn't exist or is unset).
+
+You can define a filterx statement like this:
+
+```shell
+filterx <identifier> {
+    <filterx-statement-1>;
+    <filterx-statement-2>;
+    ...
+};
+```
+
+Then use it in a log path:
+
+```shell
+log {
+    source(s1);
+    filterx(<identifier>);
+    destination(d1);
+};
+```
+
+You can also define it inline. For details, see {{% xref "/chapter-configuration-file/inline-objects/_index.md" %}}.
+
+For example, the following filterx statement selects the messages that contain the word `deny` and come from the host `example`.
+
+```shell
+filterx demo_filterx {
+    ${HOST} == "example";
+    ${MESSAGE} =~ "deny";
+};
+log {
+    source(s1);
+    filter(demo_filterx);
+    destination(d1);
+};
+```
+
+The following example does the same, but defines the filterx block inline.
+
+```shell
+log {
+    source(s1);
+    filterx demo_filterx {
+        ${HOST} == "example";
+        ${MESSAGE} =~ "deny";
+    };
+    destination(d1);
+};
+```
+
+You can use filterx blocks together with other blocks in a log path, for example, use a parser before/after the filterx block if needed.
+
+<!-- FIXME what is mutable/immutable writable/read-only > devs to write a draft  -->
+
+## Filterx statements
+
+A filterx block contains one or more filterx statements. The order of the statements is important, as they are sequentially processed.  If any of the statements is false (or results in an error), {{< product >}} drops the message from that log path.
+
+Filterx statements can be one of the following:
+
+- A comparison, for example, `$HOST == "my-host";`. This statement is true only for messages where the `$HOST` field is `my-host`. Such simple comparison statements can be the equivalents of [traditional filter functions]({{< relref "/chapter-routing-filters/filters/reference-filters/_index.md" >}}).
+- A value assignment for a [name-value pair or a local variable](#variable-scope), for example, `$my-field = "bar";`. The left-side variable automatically gets the type of the right-hand expression.
```

Comment:
does something like `a = false;` evaluate to false and thus exit?

Reply:
This was the case, but fixed by assignment being excluded from the "falsy" logic.